python/across_api/burstcube/schema.py

```diff
@@ -0,0 +1,239 @@
+# Copyright © 2023 United States Government as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# All Rights Reserved.
+
+
+import json
+from enum import Enum
+from typing import List, Optional
+
+import astropy.units as u  # type: ignore
+from astropy.time import Time  # type: ignore
+from pydantic import model_validator  # type: ignore
+
+from ..base.schema import (
+    AstropySeconds,
+    AstropyTime,
+    BaseSchema,
+    OptionalDateRangeSchema,
+    OptionalPositionSchema,
+)
+
+
+class TOOReason(str, Enum):
+    """
+    Reasons for rejecting TOO observations
+
+    Attributes
+    ----------
+    saa : str
+        In SAA
+    earth_occult : str
+        Earth occulted
+    moon_occult : str
+        Moon occulted
+    sun_occult : str
+        Sun occulted
+    too_old : str
+        Too old
+    other : str
+        Other
+    none : str
+        None
+    """
+
+    saa = "In SAA"
+    earth_occult = "Earth occulted"
+    moon_occult = "Moon occulted"
+    sun_occult = "Sun occulted"
+    too_old = "Too old"
+    other = "Other"
+    none = "None"
+
+
+class TOOStatus(str, Enum):
+    """
+    Enumeration class representing the status of a Target of Opportunity (TOO) request.
+
+    Attributes:
+    requested : str
+        The TOO request has been submitted.
+    rejected : str
+        The TOO request has been rejected.
+    declined : str
+        The TOO request has been declined.
+    approved : str
+        The TOO request has been approved.
+    executed : str
+        The TOO request has been executed.
+    other : str
+        The TOO request has a status other than the predefined ones.
```

Comment:
The quality of the contextual help in code editors may be better if you make these descriptions the docstrings of their respective enum attributes rather than of the class itself, like this:

```python
    requests = "Requested"
    """The TOO request has been submitted."""
```

Reply:
I tried this and it didn't work in VSCode. The docstrings added after the Enum values are not displayed in tooltip hovers. However, the docstring for the whole class is shown. 

This is what it looks like with the docstring:

<img width="249" alt="Screenshot 2024-02-26 at 14 23 04" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/15330708/746692c8-edce-462d-aa23-bdbe6643f808">

without:

<img width="297" alt="Screenshot 2024-02-26 at 14 27 58" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/15330708/692ab747-6066-4f80-90ca-7c340004bf07">

This is what hovering over the enum looks like in both cases:

<img width="327" alt="Screenshot 2024-02-26 at 14 39 43" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/15330708/7cf97622-3bec-424b-8eb3-b125b5b8e7e3">

Reply:
Oh well. Sphinx renders docstrings for member variables. Thanks for checking.